### ISSUE TITLE
JSON wrapper object not mapped to proper OpenAPI model

### ISSUE THREAD

**Issue by lwielek** [swagger-api/swagger-core#2884]
When a class is annotated with 
```
@com.fasterxml.jackson.annotation.JsonTypeInfo(use = com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include = com.fasterxml.jackson.annotation.JsonTypeInfo.As.WRAPPER_OBJECT)
```
it is being serialized as an object, rather than a simple string.
The OpenAPI model for such a class is represented as a simple value.

See example attached to issue #2862.

Serialized:
```
{"ExampleSubType":"VALUE1"}
```

Model:
```
Example:
  type: object
ExampleSubType:
  type: string
  allOf:
  - $ref: '#/components/schemas/Example'
  enum:
  - VALUE1
  - VALUE2
```

**Comment by jbraga** [comment]
I need this too :(
@lwielek, have you found a workaround?

**Comment by lwielek** [comment]
@jbraga 
no, my workaround was not to use wrapper object annotations ...

**Comment by sakibulalam** [comment]
any progress on this issue?

**Comment by matejuh** [comment]
This is quite painful bug which blogs us to use specification generated from code. This test: https://github.com/swagger-api/swagger-core/blob/4f30e3eb594790bdbe4f4a12553a54e2f7d8c5cd/modules/swagger-core/src/test/java/io/swagger/v3/core/converting/ModelConverterTest.java#L130 is invalid.
It should be: `employee.getProperties().get("employee").getProperties();`

**Comment by dazraf** [comment]
Is there anyone reading this, or has development on swagger stopped?

**Comment by frantuma** [comment]
@dazraf there is indeed somebody reading, and nope, development has not stopped; as always contributions are more than welcome though..

That said a partial implementation has been implemented in #3921, available in latest snapshot and next release.

**Comment by dazraf** [comment]
Thank you @frantuma!

**Comment by MichakrawSB** [comment]
Closing this issue as part of our https://github.com/swagger-api/swagger-core/issues/5074.
This issue is outdated and/or inactive for a long time, and we’re not able to revisit it meaningfully.

If this problem is still relevant, please open a new issue and reference this one for context.
Thank you for your understanding!
